### PR TITLE
Support placeholder in SofaService and SofaReference annotation.

### PR DESCRIPTION
**Pull request by QilongZhang**
#260

### CODE REVIEW

**Comment by QilongZhang** [comment]
@chpengzh  @JervyShi

**Comment by chpengzh** [comment]
这个竟然是用代理去实现的，又学到了新姿势！

File: runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/annotation/PlaceHolderBinder.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+package com.alipay.sofa.runtime.annotation;
+
+/**
+ * @author qilong.zql
+ * @since 2.5.2
+ */
+public interface PlaceHolderBinder {
+    String bind(String string);
```
Comment (straybirdzls) [inline runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/annotation/PlaceHolderBinder.java]: Please add comment for this interface and method. And should we provide a default Implement?  I see a lot of duplicate implements below.

File: runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/spring/ServiceAnnotationBeanPostProcessor.java [hunk 2]
```diff
@@ -88,10 +96,19 @@ public Object postProcessAfterInitialization(Object bean, String beanName)
...
+            }
+        };
 
-        SofaService sofaServiceAnnotation = beanClass.getAnnotation(SofaService.class);
+        AnnotationWrapperBuilder<SofaService> builder = AnnotationWrapperBuilder.wrap(
+            beanClass.getAnnotation(SofaService.class)).withBinder(binder);
+        SofaService sofaServiceAnnotation = builder.build();
 
         if (sofaServiceAnnotation == null) {
             return;
```
[the remark this answers is not in the document]
Comment (QilongZhang) [reply runtime-sofa-boot-starter/src/main/java/com/alipay/sofa/runtime/spring/ServiceAnnotationBeanPostProcessor.java]: Good advice.

**Pull request merged**